opendatahub-io/odh-dashboard
# feat: modify behaviours for the enable button for NIM.

Jira: [https://issues.redhat.com/browse/NVPE-37](https://issues.redhat.com/browse/NVPE-37)
Jira: [https://issues.redhat.com/browse/NVPE-72](https://issues.redhat.com/browse/NVPE-72)

## Description



- Show [Skeleton](https://www.patternfly.org/components/skeleton/) when visibility of the enable button is determined.
- Disable the submit button if the API key is not provided by user.
- Hide the link to remove the NIM app from the enabled apps page when NIM is disabled.
- Hide the enable button after NIM app is enabled.

<img width="1723" alt="image" src="https://github.com/user-attachments/assets/42eb7336-3b7b-4936-b37f-fed69c7cc1b0">
<img width="1724" alt="image" src="https://github.com/user-attachments/assets/01330328-eba1-49f6-8c77-8c1590b88690">
<img width="1724" alt="image" src="https://github.com/user-attachments/assets/e785f098-4748-4cf3-8235-4778a013b6a8">


## How Has This Been Tested?




- Tested locally.

## Test Impact



- Go to the `Explore` page, select the `NVIDIA NIM` tile, and should see the [Skeleton](https://www.patternfly.org/components/skeleton/)when loading. If NIM is enabled, the `Enable` button should be hidden, if NIM is disabled but can be enabled, the `Enable` button should be displayed, if NIM cannot be enabled, the `Enable` button should be displayed but the button should be disabled, hover the button, the tooltip should be displayed.
- Click the `Enable` button, and the modal to enable NIM should be displayed. If no API key is provided, the `Submit` button should be disabled, otherwise, it should be enabled.
- Provide API key, and click the `Submit` button. If validation success, the modal should be closed and the `Enable` button on the `Explore` page should be hidden.
- Disable NIM and go to the `Enable` page -> click the `disable` on the top-right corner of the NIM tile, the link to remove NIM from this page should NOT be seen.

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
